src/index.js

```diff
@@ -1,6 +1,8 @@
-import { createRoot } from 'react-dom/client'
-import { App } from './App'
+import ReactDOM from 'react-dom/client'
+import Router from './pages/Router'
 
-const container = document.getElementById('app')
-const root = createRoot(container)
-root.render(<App />)
+const rootElement = document.getElementById('app')
+if (!rootElement.innerHTML) {
+    const root = ReactDOM.createRoot(rootElement)
```

Comment:
Can you do createRoot here like before? Not React.createRoot? I'm pretty sure with React 18 they said you have to do the imported createRoot to get concurrent rendering feature.

Reply:
So changing this doesn't affect React 18, but it is still cleaner than importing ReactDOM just to use createRoot.

Further reading on why I thought this: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis